c++/src/kj/common.h

```diff
@@ -1396,6 +1396,11 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
   inline constexpr ArrayPtr(decltype(nullptr)): ptr(nullptr), size_(0) {}
   inline constexpr ArrayPtr(T* ptr, size_t size): ptr(ptr), size_(size) {}
   inline constexpr ArrayPtr(T* begin, T* end): ptr(begin), size_(end - begin) {}
+  ArrayPtr<T> operator=(Array<T>&&) = delete;
+  ArrayPtr<T> operator=(nullptr_t) {
+    ptr = nullptr;
+    size_ = 0;
+  }
```

Comment:
Missing return statement.

Note that by convention, `operator=` should always return a reference to `*this`, so the return type here needs to be `ArrayPtr<T>&`.

In `StringPtr`, you had `operator=` returning `void`, but that's not correct either -- it should return `StringPtr&`.